### ISSUE TITLE
Add objective wrapper

### ISSUE THREAD

**Issue by HenrikMettler** [Happy-Algorithms-League/hal-cgp#244]
In #238 it was discussed to have an objective wrapper. This has been removed from #238 since implementing the counter as an attribute of the wrapper did not work with multiple processes in parallel. However an objective wrapper might still be nice to have, since we could remove the `if individual.fitness is not None` from the implementations of objective functions.

**Comment by jakobj** [comment]
hmm, i'm not sure about this any more. if it's just about catching the `fitness is not None` individuals, i'd prefer to just leave it up to the user to add it, or not (e.g., for stochastic objectives it should *not* be added), seems like less cognitive overhead when first learning to use the library.
what do you think @mschmidt87 ?

**Comment by jakobj** [comment]
ping @mschmidt87 

also @HenrikMettler do you have strong opinions on this one?

**Comment by HenrikMettler** [comment]
Not really. I agree with your point on stochastic objectives, I think could become a common use case. We could close this for now and for the case that the wrapper becomes more useful, I keep the local development branch?

**Comment by jakobj** [comment]
> Not really. I agree with your point on stochastic objectives, I think could become a common use case. We could close this for now and for the case that the wrapper becomes more useful, I keep the local development branch?

sounds good to me :+1:

**Comment by mschmidt87** [comment]
Sorry, I had overlooked this discussion completely. 
As far as I can see, this decorator is useful for deterministic objective functions, but not useful for stochastic objective functions. So, what about just renaming the decorator in #245  to something like `deterministic_objective` or more descriptive `single_probe_objective`?

**Comment by jakobj** [comment]
hmm, i'm not sure. what are the benefits for the disadvantages for the user of providing such a decorator? benefit is saving two lines of boilerplate code for deterministic objectives, disadvantage is increased complexity/cognitive load. anything else?

**Comment by jakobj** [comment]
ping :bell: @mschmidt87

**Comment by mschmidt87** [comment]
Sorry for the delay. I am fine with closing this without merging. I see your points, but the additional cost of maintaining and explaining this code is probably not worth merging it.
